src/zc/guys.cpp

```diff
@@ -11599,7 +11599,10 @@ void eStalfos::charge_attack()
 			{
 				dir=ldir;
 				dashing=true;
-				step=zslongToFix(dstep*100)+1;
+				if(dmisc10<=0 || replay_version_check(0,33))
+					step=zslongToFix(dstep*100)+1;
+				else
+					step=zslongToFix(dstep*100)+(dmisc10/100);
```

Comment:
Firstly, there's a bug here- `dmisc10/100` is an integer division. So here, any value between "0" and "99" would all do the same thing! Normally in C++, you would avoid this by using floating-point (ex. `dmisc10 / 100.0`) - but for ZC we prefer to keep everything in the fixed-point (`zfix`) typing instead. That is, in fact, already what the rest of this line is doing- so it's simple to convert it to match.
```suggestion
					step=zslongToFix(dstep*100)+zslongToFix(dmisc10*100);
```
or, you can even combine the terms
```suggestion
					step=zslongToFix((dstep+dmisc10)*100);
```

Secondly, personally, I feel like the charge step being customized should *override* the base speed, rather than *add* to it - what if I want to create an enemy that darts around quickly, but then lumbers slowly towards you when it notices you? (...maybe not that necessary a use case, but still it would be simple enough to support the option)
```suggestion
					step=zslongToFix(dmisc10*100);
```